### PR TITLE
feat: Handle V3 DMs when ready 

### PR DESCRIPTION
**Pull request by alexrisch**
Added helpers
Updated naming to match v3 rather than just groups
Updated Pinned Conversations and Conversation List

### CODE REVIEW

File: components/PinnedConversations/PinnedV3Conversation.tsx [hunk 1]
```diff
@@ -1,61 +1,25 @@
...
+    return null;
+  }
+  if (conversationIsGroup(conversation)) {
     return (
-      <GroupAvatarDumb
-        size={AvatarSizes.pinnedConversation}
-        members={memberData}
-        style={{ marginLeft: 16, alignSelf: "center" }}
-      />
+      <PinnedV3GroupConversation group={conversation as GroupWithCodecsType} />
```
Comment (thierryskoda) [inline components/PinnedConversations/PinnedV3Conversation.tsx:19]: Normally we shouldn't need the `as` if the type assertion is done correctly

File: components/PinnedConversations/PinnedV3DMConversation.tsx [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+  const preferredName = usePreferredName(peer);
+  const preferredAvatar = usePreferredAvatarUri(peer);
+
+  const onLongPress = usePinnedConversationLongPress(conversation.topic);
+  const onPress = useCallback(() => {
+    navigate("Conversation", { topic: conversation.topic });
+  }, [conversation.topic]);
+
+  const title = preferredName;
+  const showUnread = false;
```
Comment (thierryskoda) [inline components/PinnedConversations/PinnedV3DMConversation.tsx:28]: unnecessary IMO

File: components/PinnedConversations/PinnedV3GroupConversation.tsx [hunk 3]
```diff
@@ -0,0 +1,66 @@
+import { PinnedConversation } from "./PinnedConversation";
```
Comment (thierryskoda) [inline components/PinnedConversations/PinnedV3GroupConversation.tsx:1]: LOVE how the 2 (groups and DMS) are separated 🙏

File: components/PinnedConversations/PinnedV3DMConversation.tsx (hunk 2 above)
Comment (alexrisch) [reply components/PinnedConversations/PinnedV3DMConversation.tsx:28]: Yeah mostly a setup for when that would exist, but just not finished work

File: components/PinnedConversations/PinnedV3Conversation.tsx (hunk 1 above)
Comment (alexrisch) [reply components/PinnedConversations/PinnedV3Conversation.tsx:19]: Yeah, I've made a PR for the SDK to resolve that: https://github.com/xmtp/xmtp-react-native/pull/525

**Pull request merged**